.github/workflows/ci.yml

```diff
@@ -3,28 +3,40 @@ name: Github Actions
 on: ["push", "pull_request"]
 
 jobs:
-  test:
+  test-linux-macos:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        python-version: ["3.9", "3.10", "3.11", "3.12", "3.13", "3.13t"]
-        architecture: [x86, x64]
+        python-version: ["3.9", "3.13", "3.13t"]
         os:
-          [
-            ubuntu-latest,
-            ubuntu-22.04,
-            macos-13,
-            windows-latest,
-            windows-2019,
-          ]
-        exclude:
-          - os: ubuntu-latest
-            architecture: x86
-          - os: ubuntu-22.04
-            architecture: x86
-          - os: macos-13
-            architecture: x86
+          # Note that macos-13 is x86-64 (deprecated already),
+          # and macos-latest is arm64.
+          [ubuntu-22.04, ubuntu-24.04-arm, macos-13, macos-latest]
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install
+        run: |
+          pip install . -v
 
+      - name: Test with pytest
+        run: |
+          pip install pytest
+          cd doc  # avoid picking up bottleneck from the source dir
+          pytest --pyargs bottleneck
+
+  test-windows:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        python-version: ["3.9", "3.13", "3.13t"]
```

Comment:
Are you sure this is enough? I was of the opinion that, as far it doesn't have an impact because they run in parallel, we might want to test all the supported python versions

Reply:
It should be. Other option: how about a separate job on only `ubuntu-latest` then that exhaustively does all Python versions? That'd be 6 jobs.

It doesn't seem reasonable to do 6 platforms x 6 python versions x 2 architectures = 72 jobs.

Reply:
Ok sounds good.